lib/versionmanager.js

```diff
@@ -241,12 +242,28 @@ function upgradePackageDefinitions(currentDependencies, options) {
  * @param {Object} [options={}]
  * @returns {string} The updated package data, as utf8 text
  */
-function upgradePackageData(pkgData, oldDependencies, newDependencies, newVersions, options = {}) {
+async function upgradePackageData(pkgData, oldDependencies, newDependencies, newVersions, options = {}) {
     for (const dependency in newDependencies) {
         if (!options.minimal || !isSatisfied(newVersions[dependency], oldDependencies[dependency])) {
+            if (options.upgrade && options.interactive) {
+                const to = versionUtil.colorizeDiff(newDependencies[dependency] || '', oldDependencies[dependency]);
+                const q = {
+                    type: 'confirm',
+                    name: 'value',
+                    message: `Do you want to upgrade: ${dependency} ${oldDependencies[dependency]} → ${to}?`,
+                    initial: true
+                };
+                let response = await prompts(q);
+                if (!response.value) {
+                    delete newDependencies[dependency];
```

Comment:
I am hesitant to mutate `oldDependencies` and `newDependencies` (mutations = bad). Is there anything downstream that depends on these being mutated here? If so, would it be possible to copy them, mutate the copy, and then pass them down instead?

Reply:
mutation of oldDependencies was unnecessary, however  newDependencies as I understand is used by reference when printing updated dependency list. 

`upgraded = newDependencies`

https://github.com/tjunnone/npm-check-updates/blob/6ba3c1d1e78c29e4c38e771a681d9a2aa46d078c/lib/npm-check-updates.js#L201-L208